src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -156,33 +159,20 @@ private static Checksum CreateUsingByteArrays(Checksum checksum1, Checksum check
 
         private static Checksum CreateUsingSpans(Checksum checksum1, Checksum checksum2)
         {
-            using var hash = s_incrementalHashPool.GetPooledObject();
-
-            Span<byte> bytesSpan = stackalloc byte[2 * HashSize];
-            Span<byte> hashResultSpan = stackalloc byte[hash.Object.HashLengthInBytes];
-
-            checksum1.WriteTo(bytesSpan);
-            checksum2.WriteTo(bytesSpan.Slice(HashSize));
+            Span<HashData> checksums = stackalloc HashData[] { checksum1._checksum, checksum2._checksum };
```

Comment:
unmanaged types can be trivially stackalloc'ed like this.  no need to stackalloc the space then manually write the data into it.